### ISSUE TITLE
ev3-print3rbot .svg files

### ISSUE THREAD

**Issue by AnastasiaPMP** [cavenel/ev3-print3rbot#16]
Thank you for your kind reply to my previous issue on this model!

Another issue is with .svg files

When I run the writer.py file, it prints only test.svg. It's ok. I still haven't figured out what I should change in the code in order to tell him to print another file, so I saved different .svg files under the same test.svg name.

It probably should have worked, but I got the following:

After I did this, I run the writer.py file again, and it gives me ERROR. Something about key code ID (I'll check the exact name and attach screens and the video of how it's working tomorrow morning').

**Comment by cavenel** [comment]
It should work indeed by changing test.svg. Else you can change line 476 of writer.py : https://github.com/cavenel/ev3-print3rbot/blob/master/writer.py#L476 and put the filename you want!
Please attach the svg as well as the screen and video so that I can see if anything is wrong.

I have a new version coming soon with easier way of changing filename.

**Comment by AnastasiaPMP** [comment]
Hi! Sorry for a bit of delay with the video and screens!

Here is the video of how it currently works: https://www.youtube.com/watch?v=s_wYLqVggBU

Here is what the picture look like each time the robot draws your test.svg file. It smudges the image:
https://drive.google.com/open?id=0B2XMSGilPq7EbFQtTUM0bkh3NE0

When I even slightly modify this file, it won't draw anything, just returns the following error message:
https://drive.google.com/open?id=0B2XMSGilPq7ESVQybXFta3NDWkk

For example, the message above is returned when I feed him with the following version of the file:
My test .svg: https://drive.google.com/open?id=0B2XMSGilPq7EaWo2aVB4YUpSN1E

**Comment by cavenel** [comment]
Ok simple correction for the svg bug (Something about key code ID) is to remove lines 349 to 352 of writer.py (https://github.com/cavenel/ev3-print3rbot/blob/master/writer.py#L349):
```python
        try:
            itemlist = filter(lambda x: x.attributes['id'].value != "borders", itemlist)
        except:
            pass
```
and it should work to load your personal svg.

**Comment by cavenel** [comment]
Can I close this issue?

**Comment by AnastasiaPMP** [comment]
Hi! Sorry for the month-long delay with the reply!

I tried your improvements.

The bugfix for the code has worked. I was able to upload any other svg file. But the image that the robot produced was mirrored.

As far as the fix for the construction is concerned, we did it, and the robot started to lift the pen but the whole construction turned out to be unstable, so the pen started to lift, but not quite properly, so it ended with the picture being smuged again because of this unstability. 

I didn't make the video at once, my bad. If you want to take a look at it, I can make on Sunday.

Are you still planning to upload the other version of the model assembling? I could try it too.
